connutils.go

```diff
@@ -615,11 +608,34 @@ func (r *configResolver) config(opts *Options) (*connConfig, error) {
 		tlsSecurity = r.tlsSecurity.val.(string)
 	}
 
+	security, err := getEnvVarSetting("EDGEDB_CLIENT_SECURITY", "default",
+		"default", "insecure_dev_mode", "strict")
+	if err != nil {
+		return nil, err
+	}
+
+	switch security {
+	case "default":
+	case "insecure_dev_mode":
+		if tlsSecurity == "default" {
+			tlsSecurity = "insecure"
```

Comment:
No-no, "default" means "default" for tlsSecurity also.

Reply:
I don't know how to interpret this. Can you write out something like a truth table for the interaction between these two values?

Reply:
```
CLIENT_SECURITY            CLIENT_TLS_SECURITY    result TLS_SECURITY
-------------------------  ---------------------  --------------------
default                    default                default
default                    insecure               insecure
default                    no_host_verification   no_host_verification
default                    strict                 strict
insecure_dev_mode          default                insecure
insecure_dev_mode          insecure               insecure
insecure_dev_mode          no_host_verification   no_host_verification
insecure_dev_mode          strict                 strict
strict                     default                strict
strict                     insecure               ERROR
strict                     no_host_verification   ERROR
strict                     strict                 strict
```

Reply:
I must be missing something. Where does the implementation differ from the table?

Reply:
"default" `CLIENT_SECURITY` leads to "insecure" TLS mode instead of "default"

Reply:
The `CLIENT_SECURITY=default` case (line `618`) has an empty body which is a noop. Are you reading it as if that case falls through to the next case? switch cases don't fall through unless there is an explicit `fallthrough` statement.

Reply:
Oh, yeah that was it, I was reading it as C. :+1: